### PR TITLE
fix: hardware wallet support

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves hardware wallet compatibility.

## How this PR fixes it

Both Ledger and Trezor wallet modules have been upgrade and the `blockExplorerUrl` is successfully passed in the `Chain[]` configuration.

This is confirmed working on Ledger Nano X (firmware version 2.0.2) and Trezor Model T (firmware version 2.5.2).

### Ledger

- Ensure that "Bling signing" is enabled.
- Connect device and ensure signing then executing and immediate execution is successful.

### Trezor

- Ensure "Safety checks" in the Trezor Suite are set to "Prompt".
- Connect device and ensure signing then executing and immediate execution is successful.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,17 +51,16 @@
     "@reduxjs/toolkit": "^1.8.2",
     "@sentry/react": "^7.8.1",
     "@sentry/tracing": "^7.8.1",
-    "@web3-onboard/coinbase": "^2.1.1",
-    "@web3-onboard/core": "^2.8.1",
-    "@web3-onboard/fortmatic": "^2.0.12",
-    "@web3-onboard/injected-wallets": "^2.2.0",
-    "@web3-onboard/keepkey": "^2.2.1",
-    "@web3-onboard/keystone": "^2.2.1",
```
Comment (iamacook) [inline package.json:59]: This was not being used.

File: src/services/onboard.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ export const createOnboard = (chainConfigs: ChainInfo[]): OnboardAPI => {
     token: cfg.nativeCurrency.symbol,
     color: cfg.theme.backgroundColor,
     publicRpcUrl: cfg.publicRpcUri.value,
-    // FIXME: add block explorer URL
+    blockExplorerUrl: new URL(cfg.blockExplorerUriTemplate.address).origin,
```
Comment (iamacook) [inline src/services/onboard.ts:29]: `new URL` removes any pathname, which we have in order to pass the `{{address}}`.

**Review by DiogoSoaress — APPROVED** [review]

**Comment by katspaugh** [comment]
> Ensure that "Blind signing" is enabled.

Ah, I remember in safe-react we specifically instructed users to do this, there was even a photo in the popup.

**Comment by iamacook** [comment]
> Ah, I remember in safe-react we specifically instructed users to do this, there was even a photo in the popup.

Exactly. We should look into trying to integrate something similar here as well.

**Comment by francovenica** [comment]
NOTE: This ticket is not merged, so I tried it in my localhost.

I have no issues creating off-chain tx with the trezor.
I have no issues creating and executing tx with ledger

Issue:
I'm not able to execute tx with the trezor. I always get a "parameter value missing" error. 
I have a Trezor model "One".

image

**Comment by iamacook** [comment]
> Issue: I'm not able to execute tx with the trezor. I always get a "parameter value missing" error.

Did you run `yarn` before testing?

**Comment by katspaugh** [comment]
Missing “value” is an onboard bug. I reported it to them a couple weeks ago.

**Comment by iamacook** [comment]
> Issue: I'm not able to execute tx with the trezor. I always get a "parameter value missing" error. I have a Trezor model "One".

Did you install the updated packages before testing?

**Comment by francovenica** [comment]
Tried, now I'm getting this error
image


I'll wait for this to be merged to test it, just in case is my local that has some missconfigurations

**Comment by iamacook** [comment]
> I'll wait for this to be merged to test it, just in case is my local that has some missconfigurations

This is now merged.

**Comment by francovenica** [comment]
I was able to create and execute tx with my trezor just fine.
I'll ask @liliya-soroka to try also since she has a different version of the device

**Comment by francovenica** [comment]
It was tested in other devices (Trezor T and ledger nano x).

Note: Trezor presented an issue regarding security, but it was determined that is something on trezors side, so nothing to do about it in this project.

**Pull request merged**